### DISCUSSION TITLE
`vercel dev` fails to serve Nuxt project when `npm run dev` works

### DISCUSSION THREAD

**Discussion by bigsee** [vercel/vercel#4703] [Help]
I have used now for some time. I uninstalled now today at the CLI's prompt and installed vercel instead.

Serving my Nuxt project using `npm run dev` (`nuxt`) runs no problem. However, I don't have access to my serverless functions. Using `vercel dev` results in what looks like an infinite load:

![Screen Shot 2020-06-23 at 18 24 54](https://user-images.githubusercontent.com/11768967/85429596-3ef5ed80-b57f-11ea-95d7-806af501c241.png)

...with the following errors in the console:

![Screen Shot 2020-06-23 at 18 28 25](https://user-images.githubusercontent.com/11768967/85429715-6fd62280-b57f-11ea-9026-0869e43d00aa.png)

I have tried:

1. Deploying the project.
2. Changing ports and host in nuxt.config.js (and then back again).
3. Restarting everything.
4. Reinstalling node_modules.
5. Checking that the default development command in Vercel's dashboard matches the command I have in my package.json (`nuxt`).
6. Removing the site.manifest (as that appears in the error above, although I don't feel like it's an issue)

Still no dice.

Have I missed something obvious?

**Comment by bigsee** [comment]
Just bumping this as it is still an issue. Can only test changes that involved serverless functions by deploying to Vercel each time, which works but makes for horrible DX and lots of wasted time. 

`npm run dev` works fine. `vercel dev` hangs.

**Comment by coetry** [comment]
Hi @bigsee . Can you provide a repository for us to take a closer look? Did you try logging your functions to see if you're running into any errors on your local machine? From your console logs, it looks like you are getting bad function responses.

**Comment by bigsee** [comment]
Hi @coetry 

Many thanks for coming back to me. 

I'm not trying to invoke any functions. I'm just trying to load a static page. With structure like this:

```
package.json
/api
/pages
    ...
    /library
    /pre-registration
    index.vue
```

After running `npm run dev` (which is an alias for `nuxt --port $PORT`) I can visit [http://localhost:8080/my-new-route](http://localhost:8080/my-new-route) with no problems.

After running `vercel dev` it points me to port 3000 and every route is stuck on a loading screen, including [http://localhost:3000/my-new-route](http://localhost:3000/my-new-route).

If I _do_ try to invoke a function while the `vercel dev` server is running (from Insomnia), they are actually still working fine with no errors. For example,

![Screen Shot 2020-07-02 at 10 48 36](https://user-images.githubusercontent.com/11768967/86337658-b3045580-bc51-11ea-9996-438a1c4ceca4.png)

![Screen Shot 2020-07-02 at 10 50 01](https://user-images.githubusercontent.com/11768967/86337758-d202e780-bc51-11ea-83e8-073b2bc2661a.png)

It's just the client that seems to be broken but it means I cannot test functions on the client side.

It is a private repo but I just invited you on Github.

**Comment by bigsee** [comment]
Hello @coetry 

Just wondering if you had any thoughts on this? I'm really struggling with the project and have something in the serverless function to debug. Cannot think of a way to reasonably do it without `vercel dev`...

**Comment by bigsee** [comment]
@coetry now I'm totally blocked from working on it. Since I cannot use `vercel dev`, I have to deploy with `vercel` every time I want to test a change. As a consequence, I've now hit the hourly builds limit on the plan. So I _literally_ cannot work. 😞

**Reply by paulogdm** [reply]
Hello Bigsee,

Please create a new ticket on https://vercel.com/support/request. Let me know the ticket number so we can prioritize accordingly. Also, if you can, please provide a reproducible example.

My apologies for the delay.

**Reply by bigsee** [reply]
@paulogdm many thanks for coming back. There is already a live ticket open (#55467). Do let me know if you need me to open a fresh one. I will also give you access to the private repo.

Thanks!

**Reply by bigsee** [reply]
@coetry @paulogdm \*bump\* still here a couple of months later guys...

**Reply by paulogdm** [reply]
Please share a minimal reproducible example in a public repository. Also, you can try the "canary" version of our CLI.

**Reply by bigsee** [reply]
Yes, I have already tried the "canary" version at @coetry's suggestion. 

I gave access to the private repository when originally requested and @coetry has confirmed his access. I'm not really sure how easily I can reproduce a project of this complexity without the original data nor how much more value this will offer in troubleshooting than direct access to the problematic repo itself.

Are there logs for the dev builds that I can access somewhere? I'm happy to debug this myself if someone points me in the right direction, but I'll need _something_ to go on...

On a well-earned short holiday from tomorrow. Just FYI if someone responds tomorrow and wonders why I have gone quiet.

**Reply by paulogdm** [reply]
Yes, you can use `vc dev --debug` for verbose logs. @coetry will certainly take a look soon.

**Reply by bigsee** [reply]
that's great - thanks! I will try that in the meantime and see if it yields any more info I can share.

**Reply by bigsee** [reply]
Amazingly, with the `--debug` flag, I think I might have found some more clues... ...here is the relevant output from each command:

`vercel dev`

<img width="1359" alt="vercel dev output" src="https://user-images.githubusercontent.com/11768967/90805575-176f9700-e31c-11ea-8ba3-28fe22631538.png">

Client is not served at localhost:3000 but hangs as reported in my initial post.

`vc dev --debug`

<img width="1381" alt="vc dev --debug output" src="https://user-images.githubusercontent.com/11768967/90808148-db3e3580-e31f-11ea-89cc-011507308aab.png">

Notice that the client appears to be served at a random port (64518). Final output still states that it is served at 3000.

Interestingly, serverless functions are available at localhost:3000 and work as expected. They do not work at localhost:64518.

The opposite is the case for the client - it is served correctly if I visit localhost:64518. But if I visit localhost:3000 as instructed by the CLI, it hangs with the following debug output (three times):

<img width="1369" alt="localhost:3000 client debug output" src="https://user-images.githubusercontent.com/11768967/90808814-e34aa500-e320-11ea-9007-0c22fcd477f0.png">

**Reply by tobiasdiez** [reply]
Did you ever manage to resolve this issue? I'm experiencing the same problem with `vuepress`.

**Comment by bigsee** [comment]
Anyone else experienced anything like this or got any leads?

**Comment by rafaelmonroy** [comment]
We found this tutorial on how to get Vercel serverless functions to run locally using Nuxt.js

Instead of using TS, we modified the files to use JS. 

https://dev.to/deeja/nuxt-js-and-vercel-functions-without-using-vercel-dev-4dgf

![image (1)](https://user-images.githubusercontent.com/53493117/114484169-8c028900-9bbe-11eb-9a56-3ca426890ced.png)
